sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeTest.java

```diff
@@ -471,6 +472,65 @@ public void showDownloadProgressTest() {
         verify(mReleaseDownloaderListener).showDownloadProgress(mActivity);
     }
 
+    @Test
+    public void doNotShowDownloadProgressTestInBackground() {
+        mockStatic(DistributeUtils.class);
+        when(DistributeUtils.getStoredDownloadState()).thenReturn(DOWNLOAD_STATE_ENQUEUED);
+
+        /* Mock that download time is bigger than packageInfo.lastUpdateTime. */
+        when(SharedPreferencesManager.getLong(eq(PREFERENCE_KEY_DOWNLOAD_TIME))).thenReturn(3L);
+
+        /* mReleaseDetails is not null and it's a mandatory update. */
+        when(DistributeUtils.loadCachedReleaseDetails()).thenReturn(mReleaseDetails);
+        when(mReleaseDetails.isMandatoryUpdate()).thenReturn(true);
+
+        resumeWorkflowWithNoOnResume();
+        verify(mReleaseDownloaderListener, never()).showDownloadProgress(mActivity);
+    }
+
+    @Test
+    public void showDownloadProgressTestInForeground() {
+        mockStatic(DistributeUtils.class);
+        when(DistributeUtils.getStoredDownloadState()).thenReturn(DOWNLOAD_STATE_ENQUEUED);
+
+        /* Mock that download time is bigger than packageInfo.lastUpdateTime. */
+        when(SharedPreferencesManager.getLong(eq(PREFERENCE_KEY_DOWNLOAD_TIME))).thenReturn(3L);
+
+        /* mReleaseDetails is not null and it's a mandatory update. */
+        when(DistributeUtils.loadCachedReleaseDetails()).thenReturn(mReleaseDetails);
+        when(mReleaseDetails.isMandatoryUpdate()).thenReturn(true);
+
+        resumeWorkflow(mActivity);
+        verify(mReleaseDownloaderListener).showDownloadProgress(mActivity);
+    }
+
+    @Test
+    public void doNotShowUpdateSetupFailedDialogInBackground() {
+        when(SharedPreferencesManager.getString(PREFERENCE_KEY_UPDATE_SETUP_FAILED_MESSAGE_KEY)).thenReturn("failed_message");
+
+        /* Trigger call. */
+        start();
+
+        /* Verify dialog. */
+        verify(mDialogBuilder, never()).create();
+        SharedPreferencesManager.remove(PREFERENCE_KEY_UPDATE_SETUP_FAILED_MESSAGE_KEY);
```

Comment:
Shouldn't be the dialog shown the next time we resume the app? If the state is deleted then the flow would just reset? This should be tested to be sure what is the current behavior if and if that behavior is ok (can force a npe in the code just to test).